pkg/crdschema/handler.go

```diff
@@ -0,0 +1,118 @@
+package crdschema
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/console/pkg/proxy"
+	"github.com/openshift/console/pkg/serverutils"
+)
+
+type CRDSchemaHandler struct {
+	k8sProxy *proxy.Proxy
+}
+
+// CRDPrinterColumnsResponse represents the response structure for printer columns
+type CRDPrinterColumnsResponse struct {
+	// PerVersion contains additional printer columns specific to each version
+	PerVersion map[string][]apiextensionsv1.CustomResourceColumnDefinition `json:"perVersion,omitempty"`
+}
+
+func NewCRDSchemaHandler(k8sProxy *proxy.Proxy) *CRDSchemaHandler {
+	return &CRDSchemaHandler{
+		k8sProxy: k8sProxy,
+	}
+}
+
+func (h *CRDSchemaHandler) HandleCRDSchema(w http.ResponseWriter, r *http.Request) {
+	// Validate HTTP method
+	if r.Method != http.MethodGet {
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Invalid method: only GET is allowed"})
+		return
+	}
+
+	// Get the CRD name from the URL path
+	crdName := strings.TrimPrefix(r.URL.Path, "/")
+	if crdName == "" {
+		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: "CRD name parameter is required"})
+		return
+	}
+
+	// Create a new request to fetch the full CRD
+	crdURL := fmt.Sprintf("/apis/apiextensions.k8s.io/v1/customresourcedefinitions/%s", crdName)
+
+	// Create a new request for the CRD
+	crdReq := httptest.NewRequest(http.MethodGet, crdURL, nil)
+	// Copy headers from original request
```

Comment:
We don't want to send all headers, particularly sensitive values like `Cookie` or `X-CSRFToken`. Instead of copying the headers, I'd specify only what we need like `Accept: application/json`.